### PR TITLE
Add Virtual Network Gateway

### PR DESCRIPTION
**Pull request by benhurjoel**
Epic : https://github.com/aztfmod/terraform-azurerm-caf/issues/163, Feature #165

### CODE REVIEW

File: networking_vngw.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  resource_group_name = module.resource_groups[each.value.resource_group_key].name
+  location            = lookup(each.value, "region", null) == null ? module.resource_groups[each.value.resource_group_key].location : local.global_settings.regions[each.value.region]
+  public_ip_addresses = local.combined_objects_public_ip_addresses
+  diagnostics         = local.diagnostics
+  client_config       = local.client_config
+  vnets               = local.combined_objects_networking
+  global_settings     = local.global_settings
+  settings            = each.value
+  tags                = try(each.value.tags, null)
```
Comment (LaurentLesle) [inline networking_vngw.tf:13]: Check the logic with the virtual machine for example.
tags are already sent to the settings through the each.value
```hcl
base_tags                        = try(local.global_settings.inherit_tags, false) ? module.resource_groups[each.value.resource_group_key].tags : {}
```

**Review by LaurentLesle — CHANGES_REQUESTED** [review]

**Comment by benhurjoel** [comment]
Thank you @LaurentLesle. Made changes as suggested.

File: networking_vngw.tf (hunk 1 above)
Comment (benhurjoel) [reply networking_vngw.tf]: Thank you @LaurentLesle. Made changes as suggested.

**Comment by benhurjoel** [comment]
Pull request https://github.com/aztfmod/terraform-azurerm-caf/pull/171 supersedes this one. Closing this PR to avoid duplicacy.

**Pull request closed without merging**